streaming_data_types/arrays_wa00.py

```diff
@@ -0,0 +1,123 @@
+from typing import Union, List, NamedTuple
+from datetime import datetime, timezone
+
+import flatbuffers
+import numpy as np
+from streaming_data_types.fbschemas.arrays_wav_00.WaveFormArray import WaveFormArray
+from streaming_data_types.fbschemas.arrays_wav_00.DType import DType
+from streaming_data_types.fbschemas.arrays_wav_00.WaveFormArray import (
+    WaveFormArrayStart,
+    WaveFormArrayAddXData,
+    WaveFormArrayAddYData,
+    WaveFormArrayAddXDataType,
+    WaveFormArrayAddYDataType,
+    WaveFormArrayAddTimestamp,
+    WaveFormArrayEnd,
+)
+from streaming_data_types.utils import check_schema_identifier
+
+FILE_IDENTIFIER = b"wa00"
+
+
+def serialise_wa00(
+    values_x_array: Union[np.ndarray, List],
+    values_y_array: Union[np.ndarray, List],
+    timestamp: datetime,
+) -> bytes:
+    builder = flatbuffers.Builder(1024)
+
+    type_map = {
+        np.dtype("uint8"): DType.uint8,
+        np.dtype("int8"): DType.int8,
+        np.dtype("uint16"): DType.uint16,
+        np.dtype("int16"): DType.int16,
+        np.dtype("uint32"): DType.uint32,
+        np.dtype("int32"): DType.int32,
+        np.dtype("uint64"): DType.uint64,
+        np.dtype("int64"): DType.int64,
+        np.dtype("float32"): DType.float32,
+        np.dtype("float64"): DType.float64,
+    }
+
+    if type(values_x_array) is str:
+        values_x_array = np.frombuffer(values_x_array.encode(), np.uint8)
+        datatype_x_array = DType.c_string
+    else:
+        datatype_x_array = type_map[values_x_array.dtype]
+
+    if type(values_y_array) is str:
+        values_y_array = np.frombuffer(values_y_array.encode(), np.uint8)
+        datatype_y_array = DType.c_string
+    else:
+        datatype_y_array = type_map[values_y_array.dtype]
+
+    # Build data
+    x_data_offset = builder.CreateNumpyVector(values_x_array.view(np.uint8))
+    y_data_offset = builder.CreateNumpyVector(values_y_array.view(np.uint8))
+
+    # Build the buffer
+    WaveFormArrayStart(builder)
+    WaveFormArrayAddXData(builder, x_data_offset)
+    WaveFormArrayAddYData(builder, y_data_offset)
+    WaveFormArrayAddXDataType(builder, datatype_x_array)
+    WaveFormArrayAddYDataType(builder, datatype_y_array)
+    if timestamp is not None:
+        WaveFormArrayAddTimestamp(builder, int(timestamp.timestamp() * 1e9))
+    WA_Message = WaveFormArrayEnd(builder)
+    builder.Finish(WA_Message, file_identifier=FILE_IDENTIFIER)
+    return bytes(builder.Output())
+
+
+wa00_t = NamedTuple(
+    "wa00",
+    (
+        ("values_x_array", np.ndarray),
+        ("values_y_array", np.ndarray),
+        ("timestamp", datetime),
+    ),
+)
```

Comment:
If I were doing this from scratch I'd use a `dataclass` instead of a `NamedTuple`.